crypto/rand/drbg_lib.c

```diff
@@ -310,12 +382,20 @@ int rand_drbg_restart(RAND_DRBG *drbg,
     }
 
     /* repair error state */
-    if (drbg->state == DRBG_ERROR)
+    if (drbg->state == DRBG_ERROR) {
         RAND_DRBG_uninstantiate(drbg);
+        /* The drbg->ctr member needs to be reinitialized before reinstantiation */
+        RAND_DRBG_set(drbg, drbg->nid, drbg->flags);
```

Comment:
I assume the reason for this is so that ctr_init() gets called again. Maybe we can call that in RAND_DRBG_instantiate()? Or move this call to the next if?

Reply:
Any comment about this?